### ISSUE TITLE
Plotting

### ISSUE THREAD

**Issue by hadim** [scijava/scijava-jupyter-kernel#73]
Following https://github.com/scijava/scijava-jupyter-kernel/issues/56

I think the future of plotting for the kernel (and probably for many other languages, Python included) is [Vega](http://vega.github.io/) and [Vega-lite](https://vega.github.io/vega-lite/).

I found a Scala library called [Vegas](https://www.vegas-viz.org/). I tried it very quickly with the kernel and it works.

![screenshot from 2017-06-23 09-25-33](https://user-images.githubusercontent.com/528003/27484100-f36fb362-57f5-11e7-96b7-3eed817d1dcb.png)

I only tried with Scala but I expect others scripting languages to work as well.

For now `plot.show` display a window with the rendered plot. In a near future, a Vega extension will be shipped with Jupyter Lab (and the Notebook as well): https://github.com/altair-viz/jupyter_vega/issues/35. That will allow any Vega JSON string to be rendered if the mimetype is correct.

We could create a converter that send to the client the JSON string (using `plot.toJSON`) with the correct mimetype.

**Comment by ctrueden** [comment]
@hadim Awesome to see that it is working!

**Comment by hadim** [comment]
Waiting for the converter, it should be possible to use `nb.displayMimetype` to force HTML to be displayed but for a reason, I don't get, it's not working...

![screenshot from 2017-06-23 09-48-46](https://user-images.githubusercontent.com/528003/27485032-2a27dd82-57f9-11e7-8896-2f8ed5a5efda.png)

(maybe it's scala related)

**Comment by hadim** [comment]
A nice TODO would be to create a notebook example in `notebooks/` that show a gallery of plots (in groovy since it's our default language).

**Comment by hadim** [comment]
- Vega renderers are coming to Jupyter Lab and Jupyter Notebook.
- The kernel can now render custom mimetype such as `application/vnd.vega.v2+json` (there is a bug for rendering but basis are here)

We are getting closer. Once rendering is shipped with the last Lab and Notebook release, it will be just a matter of writing the correct Converter from a `Vegas.Plot` object to a Vega JSON string.

**Comment by hadim** [comment]
Some news here.

The Vega package is now shipped by default with JLab. For the support in Notebook, you'll have to wait a bit. If not already done, I'll strongly suggest you give a shot to JLab. I use it daily without issue.

The last SJK commit allows using `nb.vega()` or `nb.vegalite()` to display any valid Vega JSON string as a Vega plot.

![screenshot from 2017-07-28 18-06-56](https://user-images.githubusercontent.com/528003/28738242-544a6f70-73c0-11e7-9dbe-ebcef200a47a.png)

I have also tried to use the Vegas library but I ran into multiple issues. The main one is that Vegas is written in Scala and that seems to make it pretty hard to use in the SJK. Better than a long explanation, here is a notebook that explains the issues: https://gist.github.com/hadim/8901c0bccd6ea8e1cd43fb8651df1fcb

So from here, I see two possibilities:

- Find a way to make Vegas work out-of-the-box with our various script engines. 
- Create or find another Java library that can render Vega/Vega-lite JSON plots

This plotting library (being Vegas or not) could also be used outside of the Scijava kernel.
